TFNetworkLayer.py

```diff
@@ -4073,6 +4073,46 @@ def get_out_data_from_opts(cls, n_out=None, out_type=None, sources=(), **kwargs)
     return super(CompareLayer, cls).get_out_data_from_opts(n_out=n_out, out_type=out_type, sources=sources, **kwargs)
 
 
+class SwitchLayer(LayerBase):
+  """
+  Wrapper around tf.where(). Uses three inputs: condition, true_from and false_from. The output of this layer contains elements of true_from
+  where condition is True, otherwise elements of false_from. condition has to be of dtype bool. true_from and false_from must have the same shape.
+  """
+
+  layer_class = "switch"
+
+  def __init__(self, condition, true_from, false_from, **kwargs):
+    """
+    :param LayerBase condition:
+    :param LayerBase true_from:
+    :param LayerBase false_from:
+    """
+    super(SwitchLayer, self).__init__(**kwargs)
+
+    assert(condition.output.dtype == "bool")
```

Comment:
You do not need brackets for `assert`.
Please use PyCharm. It will notify about such inconsistencies.